Tests/SyncDataProvidersTests/Bookmarks/BookmarksProviderTests.swift

```diff
@@ -815,6 +815,128 @@ internal class BookmarksProviderTests: BookmarksProviderTestsBase {
+
+        let bookmarkTree = BookmarkTree {}
+
+        context.performAndWait {
+            BookmarkUtils.prepareFoldersStructure(in: context)
+            bookmarkTree.createEntities(in: context)
+            try! context.save()
+        }
+
+        let received: [Syncable] = [
+            .rootFolder(children: ["1", "2"]), // Creates a Stub with id 2
+            .bookmark(id: "1")
+        ]
+
+        let rootFolder = try await handleInitialSyncResponse(received: received, clientTimestamp: Date(), serverTimestamp: "1234", in: context)
+        assertEquivalent(withTimestamps: false, rootFolder, BookmarkTree(lastChildrenArrayReceivedFromSync: ["1", "2"]) {
+            Bookmark(id: "1")
+            Bookmark(id: "2", isStub: true)
+        })
+
+        // Add new bookmark with id 3
+        context.performAndWait {
+            let root = BookmarkUtils.fetchRootFolder(context)!
+            let newBookmark = BookmarkEntity.makeBookmark(title: "3", url: "3", parent: root, context: context)
+            newBookmark.uuid = "3"
+            try! context.save()
+        }
+
+        let sent = try await provider.fetchChangedObjects(encryptedUsing: crypter)
+
+        // Only Root and "3" should be sent
+        XCTAssertEqual(sent.count, 2)
+
+        let sentRootData = sent.first(where: { $0.payload["id"] as? String == rootFolder.uuid })
+        XCTAssertNotNil(sentRootData)
+        let folderChanges = sentRootData?.payload["folder"] as? [String: [String: [String]]]
+        XCTAssertNotNil(folderChanges)
+
+        // We expect to send create for 3
+        XCTAssertEqual(folderChanges?["children"]?["insert"], ["3"])
+
+        // Ensure there is no removal for 2
+        XCTAssertNil(folderChanges?["children"]?["remove"])
+    }
+
+    func testThatRemoteRemovalOfStubReferenceRemovesTheStub() async throws {
+        let context = bookmarksDatabase.makeContext(concurrencyType: .privateQueueConcurrencyType)
+
+        let bookmarkTree = BookmarkTree {}
+
+        context.performAndWait {
+            BookmarkUtils.prepareFoldersStructure(in: context)
+            bookmarkTree.createEntities(in: context)
+            try! context.save()
+        }
+
+        let received: [Syncable] = [
+            .rootFolder(children: ["1", "2", "3"]), // Creates a Stub with id 2
+            .bookmark(id: "1")
+        ]
+
+        var rootFolder = try await handleInitialSyncResponse(received: received, clientTimestamp: Date(), serverTimestamp: "1234", in: context)
+        assertEquivalent(withTimestamps: false, rootFolder, BookmarkTree(lastChildrenArrayReceivedFromSync: ["1", "2", "3"]) {
+            Bookmark(id: "1")
+            Bookmark(id: "2", isStub: true)
+            Bookmark(id: "3", isStub: true)
+        })
+
+        // Simulate two kinds of "removal":
+        // - Removal only from children list.
+        // - Removal of entity and from children list.
```

Comment:
Could we please adjust the comment to make it more clear?

```suggestion
        // - "2" is only removed from children list.
        // - "3" is removed from children list and deleted.
```